src/webui/utils/api.js

```diff
@@ -1,49 +1,84 @@
 import storage from './storage';
+import {HTTP_STATUS} from '../../../src/lib/constants';
 
-class API {
-  request(url, method = 'GET', options = {}) {
-      if (!window.VERDACCIO_API_URL) {
-        throw new Error('VERDACCIO_API_URL is not defined!');
-      }
+/**
+ * CHECK FOR BASE API URL
+ */
+if (!window.VERDACCIO_API_URL) {
+  throw new Error('VERDACCIO_API_URL is not defined!');
+}
+const BASE_URL = window.VERDACCIO_API_URL;
 
-      const token = storage.getItem('token');
-      if (token) {
-        if (!options.headers) options.headers = {};
+/**
+ * Handles response according to content type
+ * @param {object} response
+ * @returns {promise}
+ */
+function handleResponseType(response) {
+  if (response.headers) {
+    const contentType = response.headers.get('Content-Type');
+    if (contentType.includes('application/json')) {
+      return Promise.all([response.ok, response.json()]);
+    }
+    // it includes all text types
+    if (contentType.includes('text/')) {
+      return Promise.all([response.ok, response.text()]);
+    }
+  }
+};
 
-        options.headers.authorization = `Bearer ${token}`;
-      }
+/**
+ * checkAuthentication if 401 then clears storage
+ */
+function checkAuthentication(response) {
+  if (response.status === HTTP_STATUS.UNAUTHORIZED) {
+    storage.removeItem('token');
```

Comment:
`'token'` and `'username' might be constants at the top, for future maintenance. 

```
const LOCAL_STORAGE_TOKEN = 'token';
const LOCAL_STORAGE_USERNAME = 'username'
```